### PR TITLE
feat: add match score component 🟢 🟢 🟢 ⚪️ ⚪️

### PR DESCRIPTION
**Pull request by thiyagaraj-eightfold**
https://www.figma.com/file/U4t894XWtt2xgDBA2rIeK9/Octuple-DS?node-id=292%3A708

### CODE REVIEW

File: src/components/MatchScore/MatchScore.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+export const MatchScore: FC<MatchScoreProps> = ({
+    className,
+    score = 0,
+    total = 5,
+    hideLabel = false,
+}) => {
+    const absTotal = Math.abs(total);
+    const matchScoreClasses = classNames(className, styles.matchScoreContainer);
+    return (
+        <div className={matchScoreClasses}>
```
Comment (thiyagaraj-eightfold) [inline src/components/MatchScore/MatchScore.tsx:15]: Is the progress bar here the right role for this component? Any thoughts @dkilgore-eightfold 
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/progressbar_role

File: src/components/MatchScore/MatchScore.types.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+export type MatchScoreProps = {
```
Comment (ychhabra-eightfold) [inline src/components/MatchScore/MatchScore.types.ts:1]:
```suggestion
export interface MatchScoreProps {
```

File: src/components/MatchScore/MatchScore.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        </div>
+    );
+};
+
+const getArrayOfSize = (n: number) => Array.from(Array(n));
+
+const MatchScoreCircle = ({
+    fill = 'empty',
+}: {
+    fill?: 'empty' | 'full' | 'half';
```
Comment (ychhabra-eightfold) [inline src/components/MatchScore/MatchScore.tsx:46]: Move this to MatchScrore.types
Create a type 
`export type FillType = 'empty' | 'full' | 'half'`

File: src/components/MatchScore/MatchScore.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+}) => (
+    <div
+        className={`${styles.matchScoreCircle} ${
+            fill === 'full' ? styles.full : ''
+        }  ${fill === 'half' ? styles.half : ''}`}
```
Comment (ychhabra-eightfold) [inline src/components/MatchScore/MatchScore.tsx:51]:
```suggestion
        className={classNames([
          styles.matchScoreCircle,
          {[styles.full]: fill === 'full'},
           {[styles.half]: fill === 'half'},
        ])}
```

File: src/components/MatchScore/MatchScore.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    hideLabel = false,
+}) => {
+    const absTotal = Math.abs(total);
+    const matchScoreClasses = classNames(className, styles.matchScoreContainer);
+    return (
+        <div
+            className={matchScoreClasses}
+            role="progressbar"
+            aria-valuenow={score}
+            aria-valuemax={total}
```
Comment (ychhabra-eightfold) [inline src/components/MatchScore/MatchScore.tsx:19]: Yeah i'm not sure progressbar is the correct role here. @dkilgore-eightfold can chime in on this

File: src/components/MatchScore/matchScore.module.scss [hunk 6]
```diff
@@ -0,0 +1,33 @@
+$indicator-size: 12px;
+
+.matchScoreContainer {
```
Comment (ychhabra-eightfold) [inline src/components/MatchScore/matchScore.module.scss:3]: Use kebab case here match-score-container. I know the guidlines say to use camelCase, will update the documentation. You can still use camelCAse in ts

File: src/components/ConfigProvider/ConfigProvider.stories.tsx [hunk 7]
```diff
@@ -9,6 +9,7 @@ import { Tab, Tabs, TabVariant } from '../Tabs';
 import { IconName } from '../Icon';
 import { CompactPicker } from 'react-color';
 import { ConfigProvider, OcThemeNames, useConfig } from './';
+import { MatchScore } from '../MatchScore/MatchScore';
```
Comment (ychhabra-eightfold) [inline src/components/ConfigProvider/ConfigProvider.stories.tsx:12]: Can you create an index.ts and export the component from that.
This import will be cleaner then. Also export it in octuple.js

**Review by ychhabra-eightfold — CHANGES_REQUESTED** [review]
Minor comments, overall looks good. lets get this merged today.

**Review by ychhabra-eightfold — APPROVED** [review]

File: src/components/MatchScore/MatchScore.tsx [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+import { classNames } from '../../shared/utilities';
+import { FillType, MatchScoreProps } from './MatchScore.types';
+
+export const MatchScore: FC<MatchScoreProps> = ({
+    className,
+    score = 0,
+    total = 5,
+    hideLabel = false,
+}) => {
+    const absTotal = Math.abs(total);
```
Comment (dkilgore-eightfold) [inline src/components/MatchScore/MatchScore.tsx:12]: nit: types

**Review by dkilgore-eightfold — APPROVED** [review]

File: src/components/MatchScore/MatchScore.tsx [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+    const matchScoreClasses = classNames(className, styles.matchScoreContainer);
+    return (
+        <div
+            className={matchScoreClasses}
+            role="progressbar"
```
Comment (dkilgore-eightfold) [inline src/components/MatchScore/MatchScore.tsx:17]: The role `progressbar` will get flagged by A11y test. Will need to use an ariaLabel prop and concatenate the values as needed.

**Review by dkilgore-eightfold — CHANGES_REQUESTED** [review]

**Review by dkilgore-eightfold — APPROVED** [review]

**Pull request merged**